server/server.go

```diff
@@ -618,12 +618,7 @@ func (server *Server) createTLSConfig(entryPointName string, tlsOption *configur
 
 func (server *Server) startServer(serverEntryPoint *serverEntryPoint, globalConfiguration configuration.GlobalConfiguration) {
 	log.Infof("Starting server on %s", serverEntryPoint.httpServer.Addr)
-	var err error
-	if serverEntryPoint.httpServer.TLSConfig != nil {
-		err = serverEntryPoint.httpServer.ServeTLS(serverEntryPoint.listener, "", "")
-	} else {
-		err = serverEntryPoint.httpServer.Serve(serverEntryPoint.listener)
-	}
+	err := serverEntryPoint.httpServer.Serve(serverEntryPoint.listener)
```

Comment:
I don't think you can replace `ServeTLS` with a `TLS.NewListener`, `ServeTLS` [is doing more](https://golang.org/src/net/http/server.go?s=84336:84411#L2731) than just calling the listener.

Reply:
We should be able to do those other things... but I am thinking hard about if there is a better way to solve the issue RE X-Forwarded-Proto...